src/waveform/renderers/allshader/waveformrenderertextured.cpp

```diff
@@ -0,0 +1,515 @@
+#include "waveform/renderers/allshader/waveformrenderertextured.h"
+
+#include <QOpenGLFramebufferObject>
+#include <QOpenGLShaderProgram>
+
+#include "moc_waveformrenderertextured.cpp"
+#include "track/track.h"
+#include "waveform/renderers/waveformwidgetrenderer.h"
+#include "waveform/waveform.h"
+
+namespace allshader {
+
+// static
+QString WaveformRendererTextured::fragShaderForType(WaveformRendererTextured::Type t) {
+    using Type = WaveformRendererTextured::Type;
+    switch (t) {
+    case Type::Filtered:
+        return QLatin1String(":/shaders/filteredsignal.frag");
+    case Type::RGB:
+        return QLatin1String(":/shaders/rgbsignal.frag");
+    case Type::Stacked:
+        return QLatin1String(":/shaders/stackedsignal.frag");
+    default:
+        break;
+    }
+    assert(false);
```

Comment:
avoid segfault during set. Use QString if ur returning a QString, otherwise it'll unnecessary do a utf8 to 16 conversion and allocation.
```suggestion
    DEBUG_ASSERT(false);
    return QStringLiteral(":/shaders/rgbsignal.frag");
```